src/lib/bo.rs

```diff
@@ -667,25 +667,16 @@ impl Databases {
         message: String,
         server_name: &String,
     ) -> String {
-        let exists = { self.pending_opps.read().unwrap().contains_key(&op_log_id) }; //To limit the scope of the read lock
```

Comment:
Locking the state here to check re-load. A few lines back make it possible to a race condition where the opp will exist here but not next block leading to a panic.
Here I am making that  impossible by using a single lock